bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/HapiCompositeConverter.java

```diff
@@ -164,15 +164,16 @@ public Object fromHapi(Object input) {
 
     if (composite instanceof IAnyResource) {
 
-      // Id element
       StructureField<HapiConverter<T>> schemaEntry = schemaIterator.next();
-      values[0] = schemaEntry.result().fromHapi(((IAnyResource) composite).getIdElement());
-      valueIndex++;
+      if (schemaEntry.fieldName().equals("id")) {
+        // Id element.
```

Comment:
ack

Reply:
ack in the sense done :), I have added the comments just before the if condition.

Reply:
I see, yeah, I saw that comment. I agree that is the root cause but that is not exactly what is happening here. I mean IIUC we have already dropped the `id` from the schema when we were translated the element-definitions to schema, right? Here we are accounting for that exception, i.e., we expect that there might be HAPI elements with `id` here but there is no corresponding `id` in the schema. Anyway, this is a minor issue, the reason I keep commenting on these is that Bunsen code is a little bit complex (and we have forked it), so it might help future us reading this :-)